scripts/publish_binary_size.js

```diff
@@ -106,28 +110,66 @@ function query(after) {
 
                 row[i + 1] = run ? +run.summary.match(/is (\d+) bytes/)[1] : undefined;
             }
-
             rows.push(row);
         }
 
         if (history.pageInfo.hasNextPage) {
             return query(history.pageInfo.endCursor);
         } else {
-            return new AWS.S3({region: 'us-east-1'}).putObject({
-                Body: zlib.gzipSync(JSON.stringify(rows.reverse())),
-                Bucket: 'mapbox',
-                Key: 'mapbox-gl-native/metrics/binary-size/data.json',
-                ACL: 'public-read',
-                CacheControl: 'max-age=300',
-                ContentEncoding: 'gzip',
-                ContentType: 'application/json'
-            }).promise();
+          
+          const latestRun = rows[0]
+          const runSizeMeasurements = latestRun.slice(0)
+
+          for (let i = 0; i < platforms.length; i++) {
+              const {platform, arch} = platforms[i];
+
+              sizeCheckInfo.push(JSON.stringify({
+                  'sdk': 'maps',
+                  'platform' : platform,
+                  'arch': arch,
+                  'size' : runSizeMeasurements[i],
+                  'created_at': `${date.getUTCFullYear()}-${date.getUTCMonth() + 1}-${date.getUTCDate()}`
+              }));
+          }
         }
     });
 }
 
 github.apps.createInstallationToken({installation_id: SIZE_CHECK_APP_INSTALLATION_ID})
-    .then(({data}) => {
-        github.authenticate({type: 'token', token: data.token});
-        return query();
+.then(({data}) => {
+    github.authenticate({type: 'token', token: data.token});
+
+    return query().then(function() {
+      new AWS.S3({region: 'us-east-1'}).putObject({
```

Comment:
The innermost `then()` block does not return a promise, which prevents the outermost block from reporting a success/fail as part of the script's execution. This will affect how the CI run reports success/failure from this step.